canonical-web-and-design/maas-ui
# Build UI only

## Done

- Update the build to only build the React client.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [A development MAAS](/HACKING.md#development-deployment)

### Running the branch

You can run this branch by:

- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

Buckle up, this is going to be a bumpy ride!
- Create a new container to build the snap in. Run the rest of the steps inside that container.
- sudo snap install maas-test-db
- sudo apt install make
- git clone https://git.launchpad.net/maas && cd maas
- make install-dependencies
- git config --file=.gitmodules submodule.src/maasui/src.url https://github.com/huwshimi/maas-ui
- git config --file=.gitmodules submodule.src/maasui/src.branch build-ui-only
- git submodule sync
- git submodule update --init --recursive --remote
- make snap-prime
- sudo snap try build/dev-snap/prime
- utilities/connect-snap-interfaces
- - This didn't work for me, I ended up having to edit `utilities/connect-snap-interfaces` and commented out the line `sudo snap connect maas:maas-cli maas-cli`.
- sudo maas init region+rack --maas-url=http://localhost:5240/MAAS --database-uri maas-test-db:///
- You should now be able to visit the URL of the MAAS (`http://[container-ip]:5240/MAAS`) and the UI should load.

## Fixes

Fixes: canonical-web-and-design/app-tribe#957.
Fixes: canonical-web-and-design/app-tribe#956.
